### PR TITLE
docs: spec v1.0.0

### LINKED ISSUE #131: Remove `privatekey:` from the protocol specification
@gkc says `privatekey:` is obsolete and is never used.

The objective of this ticket is to remove `privatekey:` documentation from the protocl specification and remove it from any regex patterns.

### LINKED ISSUE #158: Update protocol spec
- Update the protocol spec, removing/marking the deprecated items

### PR DESCRIPTION
**Pull request by xavierchanth**
closes #131 
closes #158 

**- What I did**

- Renamed / reorganized the 5 key types
  - **Breaking change**: replaced privatekey with self key.
  - The keys are in two groups:
    - Public, Self, Shared
    - `Augmented keys`: Hidden, Cached

- Reorganized verbs into Auth, CRUD, notifications, misc
- moved info/noop from beta to stable as misc verbs

**- How I did it**

**- How to verify it**

**- Description for the changelog**
docs: spec v1.0.0

### CODE REVIEW

**Comment by xavierchanth** [comment]
I am fixing these lint issues, formatter messed with spacing around comments

**Review by cpswan — DISMISSED** [review]

File: specification/at_protocol_specification.md [hunk 1]
```diff
@@ -1,37 +1,33 @@
...
+| **Subject**   | atProtocol specification                           |
+| ------------- | -------------------------------------------------- |
 | **Author(s)** | Colin Constable, Kevin Nickels, Jagannadh Vanghuri |
-| | Gary Casey, Chris Swan |
-| **Revision**  | v0.2.1 (draft) |
-| **Date**      | Dec, 5, 2023  |
-
-\
+|               | Gary Casey, Chris Swan, Xavier Chanthavong         |
+| **Revision**  | v1.0.0 (draft)                                     |
```
Comment (xavierchanth) [inline specification/at_protocol_specification.md:7]: Are we okay that I am keeping this marked as a draft version?

**Comment by xavierchanth** [comment]
I set this PR to draft to clarify some things in the PR which should be linked right above this comment.

**Comment by xavierchanth** [comment]
Ready for review

File: specification/at_protocol_specification.md [hunk 2]
```diff
@@ -40,106 +36,118 @@ An atServer should have 4 major sub components:
...
 
-2. Private Key
+2. Self Key
 
-   * A private key is a key which cannot be looked up any atSign user other
-   than the one created it.
-   * A private key should not be returned in a _scan_ verb result.
-   * Format of the private key should be **privatekey::<@sign>**.
+   - A self key is a key which cannot be looked up any atSign user other than
+     the one created it.
```
Comment (gkc) [inline specification/at_protocol_specification.md]: "the one **who** created it"

**Review by gkc — DISMISSED** [review]

File: specification/at_protocol_specification.md (hunk 1 above)
Comment (gkc) [reply specification/at_protocol_specification.md:7]: Yes ... we should schedule a read-through on the architecture call some day to gather final edits, catch more things we missed etc and do another PR for those, at which point I think we can remove the `(draft)` annotation

**Review by gkc — APPROVED** [review]

**Pull request merged**